pkg/ansible/proxy/proxy.go

```diff
@@ -238,15 +242,31 @@ func addWatchToController(owner kubeconfig.NamespacedOwnerReference, cMap *contr
 			awMap.Store(resource.GroupVersionKind())
 			typeString := fmt.Sprintf("%v.%v", owner.Kind, ownerGV.Group)
 			log.Info("Watching child resource", "kind", resource.GroupVersionKind(), "enqueue_annotation_type", typeString)
-			err = contents.Controller.Watch(&source.Kind{Type: resource}, &osdkHandler.EnqueueRequestForAnnotation{Type: typeString})
+			err = addWatch(contents.Controller, &source.Kind{Type: resource}, &osdkHandler.EnqueueRequestForAnnotation{Type: typeString})
 			if err != nil {
+				log.Error(err, "gvk", resource.GroupVersionKind())
 				return err
 			}
 		}
 	}
 	return nil
 }
 
+func addWatch(c controller.Controller, s source.Source, eh handler.EventHandler, predicates ...predicate.Predicate) error {
+	errChan := make(chan error, 1)
+	go func() {
+		err := c.Watch(s, eh, predicates...)
+		errChan <- err
+	}()
+
+	select {
+	case watchErr := <-errChan:
+		return watchErr
+	case <-time.After(3 * time.Second):
```

Comment:
nit: it always warms my heart when things like this are a constant defined near the top. Not sure how relevant that is here.